# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1365054657 14400
#      Thu Apr 04 01:50:57 2013 -0400
# Node ID 8ff9006cedc33ac93214244c62536f2c094638a6
# Parent  ad052cdc89ad4088b358741b0840dc57349105c7
delete unused functions and signal for updating dbstop marker

* main-window.h, main-window.cc
(main_window::handle_update_dbstop_marker_request): Delete.
(main_window::construct): Delete signal connection from
_octave_qt_event_listener::update_dbstop_marker_signal to
main_window::handle_update_dbstop_marker_request.
* octave-qt-event-listener.h, octave-qt-event-listener.cc
(octave_qt_event_listener::update_dbstop_marker): Delete.
(octave_qt_event_listener::update_dbstop_marker_signal):
Delete signal.
* octave-event-listener (octave_event_listener::update_dbstop_marker):
Delete.

diff --git a/libgui/src/main-window.cc b/libgui/src/main-window.cc
--- a/libgui/src/main-window.cc
+++ b/libgui/src/main-window.cc
@@ -506,25 +506,16 @@ void
 main_window::handle_delete_debugger_pointer_request (const QString& file, int line)
 {
 #ifdef HAVE_QSCINTILLA
   _file_editor->handle_delete_debugger_pointer_request (file, line);
 #endif
 }
 
 void
-main_window::handle_update_dbstop_marker_request (bool insert,
-                                                  const QString& file, int line)
-{
-#ifdef HAVE_QSCINTILLA
-  _file_editor->handle_update_dbstop_marker_request (insert, file, line);
-#endif
-}
-
-void
 main_window::debug_continue ()
 {
   octave_link::post_event (this, &main_window::debug_continue_callback);
 }
 
 void
 main_window::debug_step_into ()
 {
@@ -1187,21 +1178,16 @@ main_window::construct ()
   connect (_octave_qt_event_listener,
            SIGNAL (insert_debugger_pointer_signal (const QString&, int)), this,
            SLOT (handle_insert_debugger_pointer_request (const QString&, int)));
 
   connect (_octave_qt_event_listener,
            SIGNAL (delete_debugger_pointer_signal (const QString&, int)), this,
            SLOT (handle_delete_debugger_pointer_request (const QString&, int)));
 
-  connect (_octave_qt_event_listener,
-           SIGNAL (update_dbstop_marker_signal (bool, const QString&, int)),
-           this,
-           SLOT (handle_update_dbstop_marker_request (bool, const QString&, int)));
-
   // FIXME -- is it possible to eliminate the event_listenter?
 
   _octave_qt_link = new octave_qt_link ();
 
   connect (_octave_qt_link,
            SIGNAL (update_dbstop_marker_signal (bool, const QString&, int)),
            _file_editor,
            SLOT (handle_update_dbstop_marker_request (bool, const QString&, int)));
diff --git a/libgui/src/main-window.h b/libgui/src/main-window.h
--- a/libgui/src/main-window.h
+++ b/libgui/src/main-window.h
@@ -105,17 +105,16 @@ public slots:
   void handle_workspace_visible (bool);
   void handle_editor_visible (bool);
   void handle_documentation_visible (bool);
 
   void handle_entered_debug_mode ();
   void handle_quit_debug_mode ();
   void handle_insert_debugger_pointer_request (const QString& file, int line);
   void handle_delete_debugger_pointer_request (const QString& file, int line);
-  void handle_update_dbstop_marker_request (bool insert, const QString& file, int line);
   void debug_continue ();
   void debug_step_into ();
   void debug_step_over ();
   void debug_step_out ();
   void debug_quit ();
 
   void read_settings ();
   void write_settings ();
diff --git a/libgui/src/octave-qt-event-listener.cc b/libgui/src/octave-qt-event-listener.cc
--- a/libgui/src/octave-qt-event-listener.cc
+++ b/libgui/src/octave-qt-event-listener.cc
@@ -63,25 +63,16 @@ octave_qt_event_listener::insert_debugge
 void
 octave_qt_event_listener::delete_debugger_pointer (const std::string& file,
                                                    int line)
 {
   emit delete_debugger_pointer_signal (QString::fromStdString (file), line);
 }
 
 void
-octave_qt_event_listener::update_dbstop_marker (bool insert,
-                                                const std::string& file,
-                                                int line)
-{
-  emit update_dbstop_marker_signal (insert, QString::fromStdString (file),
-                                    line);
-}
-
-void
 octave_qt_event_listener::about_to_exit ()
 {
   qApp->quit ();
 }
 
 void
 octave_qt_event_listener::entered_debug_mode ()
 { emit entered_debug_mode_signal (); }
diff --git a/libgui/src/octave-qt-event-listener.h b/libgui/src/octave-qt-event-listener.h
--- a/libgui/src/octave-qt-event-listener.h
+++ b/libgui/src/octave-qt-event-listener.h
@@ -34,26 +34,24 @@ class octave_qt_event_listener
   public:
   octave_qt_event_listener (QObject *parent = 0);
 
   void current_directory_has_changed (const std::string& directory);
   void update_workspace (void);
   void update_history (void);
   void insert_debugger_pointer (const std::string& file, int line);
   void delete_debugger_pointer (const std::string& file, int line);
-  void update_dbstop_marker (bool insert, const std::string& file, int line);
   void about_to_exit ();
 
   void entered_debug_mode ();
   void quit_debug_mode ();
 
 signals:
   void current_directory_has_changed_signal (const QString& directory);
   void update_workspace_signal (void);
   void update_history_signal (void);
   void insert_debugger_pointer_signal (const QString& file, int line);
   void delete_debugger_pointer_signal (const QString& file, int line);
-  void update_dbstop_marker_signal (bool insert, const QString& file, int line);
   void entered_debug_mode_signal ();
   void quit_debug_mode_signal ();
 };
 
 #endif // OCTAVEQTEVENTLISTENER_H
diff --git a/libinterp/interp-core/octave-event-listener.h b/libinterp/interp-core/octave-event-listener.h
--- a/libinterp/interp-core/octave-event-listener.h
+++ b/libinterp/interp-core/octave-event-listener.h
@@ -41,18 +41,15 @@ class octave_event_listener
     update_history (void) = 0;
 
     virtual void
     insert_debugger_pointer (const std::string& file, int line) = 0;
 
     virtual void
     delete_debugger_pointer (const std::string& file, int line) = 0;
 
-    virtual void
-    update_dbstop_marker (bool insert, const std::string& file, int line) = 0;
-
     virtual void about_to_exit () = 0;
 
     virtual void entered_debug_mode () = 0;
     virtual void quit_debug_mode () = 0;
 };
 
 #endif // OCTAVEEVENTLISTENER_H
